src/gluonts/shell/serve.py

```diff
@@ -207,4 +207,73 @@ def predictor_factory(request) -> Predictor:
         },
     )
 
-    gunicorn_app.run()
+    return gunicorn_app
+
+
+class ServerFacade:
+    """
+    A convenience wrapper for sending requests and handling responses to
+    an inference server located at the given address.
+    """
+
+    @validated()
+    def __init__(self, base_address: str) -> None:
```

Comment:
@jaheba: I was wondering whether there is a better type than `str` to model the `base_address` value?

Reply:
I was hoping to get something standard and similar to `IPv4Address` (with the addition of DNS addresses such as `example.com`).

`NewType` or a custom `WebAddress` class will do the trick, but I will have to introduce it within the `gluonts` namespace and force users to import and use it to annotate parameters bound to the `ServerFacade` initializer.

In the absence of a standard solution, I suggest to go with the broader `str` type.